### PR TITLE
Automate Meshnet peer list tests

### PR DESCRIPTION
**Pull request by ggediminass**

### CODE REVIEW

File: test/qa/lib/meshnet.py [hunk 1]
```diff
@@ -137,6 +175,9 @@ def get_internal_peer(self) -> Peer | None:
     def get_all_internal_peers(self) -> list[Peer]:
         return self.internal_peers
 
+    def set_internal_peer_list_empty(self):
```
Comment (piotrjurkiewicz) [inline test/qa/lib/meshnet.py:178]: Maybe `clear_internal_peer_list()` will be better?

File: test/qa/lib/meshnet.py [hunk 2]
```diff
@@ -149,6 +190,56 @@ def get_external_peer(self) -> Peer | None:
     def get_all_external_peers(self) -> list[Peer]:
         return self.external_peers
 
+    def set_external_peer_list_empty(self):
```
Comment (piotrjurkiewicz) [inline test/qa/lib/meshnet.py:193]: The same.

File: test/qa/lib/meshnet.py [hunk 3]
```diff
@@ -111,6 +112,43 @@ def get_peer_name(self, name_type: PeerName) -> str:
...
+    def _convert_to_str(value):
+        return "enabled" if value else "disabled"
+
+    def to_str(self):
+        output = [
+            f"Hostname: {self.hostname}",
+            f"Nickname: {self.nickname}",
+        ]
+
+        if self.status is not None:
```
Comment (mariusSincovici) [inline test/qa/lib/meshnet.py:125]: why this is printed only when status is not `None`, same for bellow?

File: test/qa/test_meshnet.py [hunk 4]
```diff
@@ -668,3 +668,65 @@ def test_set_defaults_when_connected_2nd_set(tech, proto, obfuscated):
     assert "Status: Disconnected" in sh.nordvpn.status()
 
     assert settings.app_has_defaults_settings()
+
+
+def base_test_peer_list(filter_list: list[str] = None) -> None:
+    sh.nordvpn.mesh.peer.list()
```
Comment (mariusSincovici) [inline test/qa/test_meshnet.py:674]: is this needed? because it will be again executed on line 677

File: test/qa/test_meshnet.py [hunk 5]
```diff
@@ -668,3 +668,65 @@ def test_set_defaults_when_connected_2nd_set(tech, proto, obfuscated):
...
+
+    if "External Peers:" in "\n".join(local_formed_list) and "External Peers:\n[no peers]" not in "\n".join(local_formed_list):
+        assert "Status: connected" in local_formed_list
+
+
+@pytest.mark.parametrize("external", [True, False], ids=lambda value: "external" if value else "")
+@pytest.mark.parametrize("internal", [True, False], ids=lambda value: "internal" if value else "")
+@pytest.mark.parametrize("offline", [True, False], ids=lambda value: "offline" if value else "")
+@pytest.mark.parametrize("online", [True, False], ids=lambda value: "online" if value else "")
+@pytest.mark.flaky(reruns=3, reruns_delay=20)
```
Comment (mariusSincovici) [inline test/qa/test_meshnet.py:709]: do you know why the tests needs to be rerun? what causes the failure? 
Same for the next one.

Comment (ggediminass) [reply test/qa/test_meshnet.py:709]: Sometimes status of peer shows up as Disconnected, therefore I in introduced re-runs

File: test/qa/lib/meshnet.py (hunk 3 above)
Comment (mariusSincovici) [reply test/qa/lib/meshnet.py:125]: ok, so same function is used for `this device` and peers.

**Review by mariusSincovici — APPROVED** [review]

**Review by keliramu — APPROVED** [review]

**Pull request merged**